src/utils/function.js

```diff
@@ -2,25 +2,39 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
+// @flow
 import { findClosestFunction } from "./ast";
 import { correctIndentation } from "./indentation";
+import type { SymbolDeclarations } from "../workers/parser";
+import type { Source } from "../types";
+export function findFunctionText(
+  line: number,
+  source: Source,
+  symbols: ?SymbolDeclarations
+): ?string {
+  let func;
 
-export function findFunctionText(line, source, symbols) {
-  const func = findClosestFunction(symbols, {
-    line,
-    column: Infinity
-  });
-  if (!func) {
+  if (symbols) {
+    func = findClosestFunction(symbols, {
+      sourceId: source.id,
+      line,
+      column: Infinity
+    });
+  } else {
+    return null;
+  }
```

Comment:
small suggestion here. I think i'd prefer putting the guard statement in `findClosestFunction`. We prefer safer calls than checks before the calls

```diff
diff --git a/src/utils/ast.js b/src/utils/ast.js
index 56c30f3..6464345 100644
--- a/src/utils/ast.js
+++ b/src/utils/ast.js
@@ -104,17 +104,25 @@ function findClosestofSymbol(declarations: any[], location: Location) {
 }
 
 export function findClosestFunction(
-  symbols: SymbolDeclarations,
+  symbols: ?SymbolDeclarations,
   location: Location
 ) {
+  if (!symbols) {
+    return null;
+  }
+
   const { functions } = symbols;
   return findClosestofSymbol(functions, location);
 }
 
 export function findClosestClass(
-  symbols: SymbolDeclarations,
+  symbols: ?SymbolDeclarations,
   location: Location
 ) {
+  if (!symbols) {
+    return null;
+  }
+
   const { classes } = symbols;
   return findClosestofSymbol(classes, location);
 }
```